### ISSUE TITLE
##[error]Bad credentials

### ISSUE THREAD

**Issue by Schneegans** [svenstaro/upload-release-action#1]
We're trying to use this action in [CosmoScout VR](https://github.com/cosmoscout/cosmoscout-vr/tree/feature/actions). While it works very nicely for our workflows running on Linux, it somehow fails with `##[error]Bad credentials` on a Windows host.

Here is the error: https://github.com/cosmoscout/cosmoscout-vr/runs/234662696#step:7:8

Here is the according workflow file: https://github.com/cosmoscout/cosmoscout-vr/blob/feature/actions/.github/workflows/release.yml

Do you have an idea why this might be happening?

**Comment by svenstaro** [comment]
Might be a temporary GitHub actions flake. I'm using this action on multiple Windows, OSX, and Linux machines and it works just fine. The credentials are provided by GitHub itself during the build.

**Comment by Schneegans** [comment]
Thanks for the quick reply! Another difference between our workflow jobs are the build times: Linux takes only 30 min while Windows requires around 1.5 h. Can it be possible that the credentials provided by GitHub expire to quickly?

**Comment by svenstaro** [comment]
That's a good suspicion. Maybe ask GitHub support about that?

**Comment by Schneegans** [comment]
Done. I'll update this as soon as I get a reply!

**Comment by svenstaro** [comment]
Thanks. Now I'm also curious as to what the problem might be.

**Comment by svenstaro** [comment]
Did you ever find out what this was?

**Comment by Schneegans** [comment]
I did not get a reply yet. I'll wait some further days...

**Comment by svenstaro** [comment]
Can you try artificially shortening your build time somehow and see whether
uploading works then?

**Comment by Schneegans** [comment]
Yes, it seems to be working with reduced build times.

Do you know of any place to report issues related to GitHub actions? Some people seem to report general issues in this project: https://github.com/actions/starter-workflows/issues

**Comment by svenstaro** [comment]
I'd report it to the community forums for github actions. The github devs
are pretty active there.

**Comment by svenstaro** [comment]
Is this still an issue? I'd like to close it since I'm pretty sure it's not a problem of this Action.

**Comment by Schneegans** [comment]
Yes, it's still an issue, but we can close this anyways. If I learn something new, I'll post this here!

**Comment by Schneegans** [comment]
While I did not get any response yet, I found an [official documentation on this issue](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#about-the-github_token-secret)!

> The installation access token expires after 60 minutes. GitHub fetches a token for each job, before the job begins.

I wonder why I did not find this earlier...

So I will use a personal access token instead of `secrets.GITHUB_TOKEN`. This should work, I guess.
